### PR TITLE
Create unit-test helper functions

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

Create new unittest helper to verify specific part of a PodTemplate, DS, EDS

### Motivation

easy unit-tests maintenance

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: controllers/datadogagent/testutils/podtemplate_utils.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		}
+		t.Errorf("EnvVar %s not founded in container %s", envVar.Name, container.Name)
+	}
+	return check
+}
+
+// CheckEnvFromIsPresent used to check if an corev1.EnvFromSource is present in a corev1.Container object
+func CheckEnvFromIsPresent(envFrom *corev1.EnvFromSource) CheckContainerFunc {
+	check := func(t *testing.T, container *corev1.Container) {
+		for _, envVarIn := range container.Env {
```
Comment (clamoriniere) [inline controllers/datadogagent/testutils/podtemplate_utils.go:115]:
```suggestion
		for _, envVarIn := range container.EnvFrom {
```

File: controllers/datadogagent/testutils/metadata_utils.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+func CheckAnnotationIsPresent(key, value string) CheckObjectMeta {
+	check := func(t *testing.T, obj *metav1.ObjectMeta) {
+		checkIsPresentInMap(t, key, value, obj.Annotations, "obj.Annotations")
+	}
+
+	return check
+}
+
+// CheckLabelIsNotPresent used to check if a label (key,value) is not present on a metav1.ObjectMeta object.
+func CheckLabelIsNotPresent(key string) CheckObjectMeta {
```
Comment (vboulineau) [inline controllers/datadogagent/testutils/metadata_utils.go:46]: As the functions return the `CheckObjectMeta` meta type, we cannot easily combine them, e.g. wrapping `CheckLabelIsPresent` in a `not` or something, meaning that we'll have to provide a helper each time (regexp, not regexp, etc.)

Comment (clamoriniere) [reply controllers/datadogagent/testutils/metadata_utils.go:46]: Do you see a solution?

Comment (vboulineau) [reply controllers/datadogagent/testutils/metadata_utils.go:46]: Maybe we could change to `func Check(runtime.Object) bool` and use a typed struct to inject other parameters.
Something like:
```
CheckLabelIsPresent{label: <some_key>}
with func (*CheckLabelIsPresent) Check(runtime.Object) bool
```

Comment (clamoriniere) [reply controllers/datadogagent/testutils/metadata_utils.go]: @vboulineau done

**Pull request merged**